homeassistant/components/assist_satellite/entity.py

```diff
@@ -0,0 +1,141 @@
+from homeassistant.helpers import entity
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.util import ulid
+
+from .models import AssistSatelliteState
+
+_CONVERSATION_TIMEOUT_SEC: Final = 5 * 60  # 5 minutes
+
+
+class AssistSatelliteEntity(entity.Entity):
+    """Entity encapsulating the state and functionality of an Assist satellite."""
+
+    entity_description = EntityDescription(
+        key="assist_satellite",
+        translation_key="assist_satellite",
+        entity_category=EntityCategory.CONFIG,
+    )
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_should_poll = False
+    _attr_state: AssistSatelliteState | None = AssistSatelliteState.LISTENING_WAKE_WORD
+
+    _conversation_id: str | None = None
+    _conversation_id_time: float | None = None
+
+    _run_has_tts: bool = False
+
+    async def _async_accept_pipeline_from_satellite(
+        self,
+        audio_stream: AsyncIterable[bytes],
+        start_stage: PipelineStage = PipelineStage.STT,
+        end_stage: PipelineStage = PipelineStage.TTS,
+        pipeline_entity_id: str | None = None,
+        vad_sensitivity_entity_id: str | None = None,
+        wake_word_phrase: str | None = None,
+    ) -> None:
+        """Triggers an Assist pipeline in Home Assistant from a satellite."""
+        pipeline_id: str | None = None
+        vad_sensitivity = vad.VadSensitivity.DEFAULT
+
+        if pipeline_entity_id:
+            # Resolve pipeline by name
+            pipeline_entity_state = self.hass.states.get(pipeline_entity_id)
+            if (pipeline_entity_state is not None) and (
+                pipeline_entity_state.state != OPTION_PREFERRED
+            ):
+                for pipeline in async_get_pipelines(self.hass):
+                    if pipeline.name == pipeline_entity_state.state:
+                        pipeline_id = pipeline.id
+                        break
+
+        if vad_sensitivity_entity_id:
+            vad_sensitivity_state = self.hass.states.get(vad_sensitivity_entity_id)
+            if vad_sensitivity_state is not None:
+                vad_sensitivity = vad.VadSensitivity(vad_sensitivity_state.state)
+
+        device_id: str | None = None
+        if self.registry_entry is not None:
+            device_id = self.registry_entry.device_id
+
+        # Reset conversation id, if necessary
+        if (self._conversation_id_time is None) or (
+            (time.monotonic() - self._conversation_id_time) > _CONVERSATION_TIMEOUT_SEC
+        ):
+            self._conversation_id = None
+
+        if self._conversation_id is None:
+            self._conversation_id = ulid.ulid()
+
+        # Update timeout
+        self._conversation_id_time = time.monotonic()
+
+        # Set entity state based on pipeline events
+        self._run_has_tts = False
+
+        await async_pipeline_from_audio_stream(
+            self.hass,
+            context=self._context or Context(),
```

Comment:
`self._context` can be too old, copy this logic: https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L1201-L1206 (maybe that should become a property?)